src/middleware/authentication.js

```diff
@@ -5,7 +5,7 @@ const auth = async (req, res, next) => {
   const authHeader = req.headers.authorization;
 
   if (!authHeader || !authHeader.startsWith('Bearer ')) {
-    throw new UnauthenticatedError('Authentication invalid');
+    return next(new UnauthenticatedError('Authentication invalid'));
```

Comment:
We might want to add a expiresIn: '2h' to JWTs, if so, we can check for expiration like so. 

const jwt = require('jsonwebtoken');
const { UnauthenticatedError } = require('../errors');

const auth = async (req, res, next) => {
  const authHeader = req.headers.authorization;

  if (!authHeader || !authHeader.startsWith('Bearer ')) {
    return next(new UnauthenticatedError('Authentication invalid'));
  }

  const token = authHeader.split(' ')[1];
  try {
    const payload = jwt.verify(token, process.env.JWT_SECRET);
    req.user = { userId: payload.userId, firstName: payload.firstName };
    next();
  } catch (error) {
    if (error.name === 'TokenExpiredError') {
      return next(new UnauthenticatedError('Token has expired. Please log in again.'));
    }
    return next(new UnauthenticatedError('Authentication invalid'));
  }
};

module.exports = auth;

Reply:
and this where we are creating them: 
const jwt = require('jsonwebtoken');

const createJWT = (user) => {
  return jwt.sign(
    { userId: user._id, firstName: user.firstName },
    process.env.JWT_SECRET,
    { expiresIn: '2h' } 
  );
};

Reply:
We've already have this in the user model, so I will just change the token lifetime in `.env`